### DISCUSSION TITLE
npm install not working

### DISCUSSION THREAD

**Discussion by tariq-ansari** [Sunbird-Ed/Community#9] [Installation] [answered]
Hi @sharathkashyap , We have been trying to build the sunbird app. Below is the configuration.

<img width="1019" alt="Screenshot 2022-05-24 at 7 27 28 PM" src="https://user-images.githubusercontent.com/68503506/170053596-a86eed5d-aca9-4eff-a737-708f86b71b39.png">

However, npm install gets stuck while install ionic core every time.  Below is the preview.
<img width="1792" alt="Screenshot 2022-05-24 at 7 22 33 PM" src="https://user-images.githubusercontent.com/68503506/170052495-9c42097d-a2fe-4a19-ac6a-7b1c284b1ddd.png">

Please advice how to proceed.

**Comment by sharathkashyap** [comment, marked as answer]
@tariq-ansari It takes unusally long time to get npm install working on ionic-core. It gets succeeded after few mins. Happens mostly due to node issue. Following things can be tried out. 
a) Use nvm to switch one version lower(i.e node 12) and reinstall cordova, ionic and  rerun the installation.
(or)
b) Uninstall node completely and freshly install node and do the mobile app installation. 

Either of these approaches can be used.
cc: @luckynvdu

**Reply by sharathkashyap** [reply]
Try this. (If you are using MAC)
first run "npm cache clean --force"
then run "npm i"

**Reply by sharathkashyap** [reply]
@tariq-ansari I had tried fresh installation. Your **concern is quite valid**. For the first time installation, It does take quite some time to get everything installed in node_modules. Mine almost took 10+ minutes to complete npm installation. I was able to complete the build at the end of 20+ minutes. This is for first time installation. We will certainly work towards improving the build time. Try rebuilding apk you should be able to complete installation in 25 mins.

**Reply by sharathkashyap** [reply]
@tariq-ansari Kindly let us know if you are able to proceed further. I will mark the answer and lock the discussion if the issue is resolved.
cc: @luckynvdu

**Reply by tariq-ansari** [reply]
> Try this. (If you are using MAC) first run "npm cache clean --force" then run "npm i"

@sharathkashyap This helped us to complete the npm installation. 
However we are getting error while executing the build script. Starting a new thread for that one.

**Reply by sharathkashyap** [reply]
@tariq-ansari Glad the issue got resolved. Will be closing the activity further.
